api/src/pcapi/core/offers/repository.py

```diff
@@ -723,26 +723,35 @@ def is_non_free_offer_subquery(offer_id: int) -> sa.sql.selectable.Exists:
     )
 
 
-def get_offer_by_id(offer_id: int) -> models.Offer:
+def get_offer_by_id(offer_id: int, load_options: typing.Optional[typing.List[str]] = None) -> models.Offer:
```

Comment:
[détail / style] Rien de bloquant : 

- On utilise `x | None` plutôt que `Optional[x]`. C'est plus concis et ça reste lisible.
- `typing.List` n'est plus nécessaire : on peut utiliser le built-in : `list[str]`
- On pourrait typer comme suit : `load_options: Iterable[str] = ()` qui permet d'avoir une valeur par défaut du bon type (mais pas une liste qui serait mutable et donc dangereux). Ça évite de se poser la question "y a-t-il une différence entre passer une liste vide et passer `None` ?" Et ça évite la ligne du dessous.

En bonus, on pourrait même essayer de détecter les mauvaises valeurs passées à `load_options`, avec : 
1. `list[typing.Literal["stock"] | typing.Literal["mediations"] | ...]`.
2. ou avec `list[GetOfferByIdLoad]`, où `GetOfferByIdLoad` serait une enum.

Je ne suis pas forcément fan de l'enum (c'est un peu lourd pour juste une fonction)... mais utiliser 6 variantes de `typing.Litteral["xxx"]` va rendre le typage bien plus long... Avis tiers bienvenus !

Reply:
Si on combine les deux idées, ça peut donner ça:

```python
# offers/repository.py

OfferLoadOptions = list[Literal["stock", "mediations", "product", "price_category", "venue", "bookings_count"]]

...

def get_offer_by_id(offer_id: int, load_options: OfferLoadOptions) -> models.Offer:
   ...

# ailleurs dans le repo

offer = offers_repository.get_offer_by_id(offer_id, load_options=["toto"])
```

```
.venv ❯ mypy src
src/pcapi/routes/pro/offers.py:69:75: error: List item 0 has incompatible type "Literal['toto']"; expected "Literal['stock', 'mediations', 'product', 'price_category', 'venue', 'bookings_count']"  [list-item]
```

: )

Reply:
J'aime bien load_options: Iterable[str] = () pour ma part

Reply:
Mais le tuple vide va fonctionner que si on a un Iterable[str] non ? Je vois pas trop comment typer load_options facilement pour avoir à la fois la vérification des valeurs ET la valeur par défaut vide qui ne soit pas un []

Reply:
`OfferLoadOptions = Iterable[Literal["stock", "mediations", "product", "price_category", "venue", "bookings_count"]]` ne fonctionne pas ?